src/relax/op/nn/convolution.cc

```diff
@@ -134,6 +134,44 @@ StructInfo InferStructInfoConv2d(const Call& call, const BlockBuilder& ctx) {
   return TensorStructInfo(ShapeExpr(out_shape), out_dtype);
 }
 
+InferLayoutOutput InferLayoutConv2d(const Call& call,
+                                    const Map<String, Array<String>>& desired_layouts,
+                                    VarLayoutMap var_layout_map) {
+  const auto& it = desired_layouts.find("relax.nn.conv2d");
+  const auto* attrs = call->attrs.as<Conv2DAttrs>();
+  ICHECK(attrs) << "Invalid Call";
+
+  Layout data_layout, weight_layout, output_layout;
+  ObjectPtr<Conv2DAttrs> new_attrs = make_object<Conv2DAttrs>(*attrs);
+
+  if (it != desired_layouts.end()) {
+    // We have a desired layout for conv2d.
+    data_layout = TransposeLike(InitialLayout(4), attrs->data_layout, (*it).second[0]);
+    weight_layout = TransposeLike(InitialLayout(4), attrs->kernel_layout, (*it).second[1]);
+    output_layout = TransposeLike(InitialLayout(4),
+                                  attrs->out_layout == "" ? attrs->data_layout : attrs->out_layout,
+                                  (*it).second.size() == 3 ? (*it).second[2] : (*it).second[0]);
+    new_attrs->data_layout = (*it).second[0];
+    new_attrs->kernel_layout = (*it).second[1];
+    new_attrs->out_layout = (*it).second.size() == 3 ? (*it).second[2] : (*it).second[0];
+  } else {
+    // We don't have a desired layout for conv2d.
+    // We can just propagate the layout from the input.
+    data_layout = GetLayout(var_layout_map, call->args[0]);
+    weight_layout = GetLayout(var_layout_map, call->args[1]);
+    output_layout = data_layout;
+    new_attrs->data_layout =
```

Comment:
It seems confusing to use the word "layout" for both data_layout and attrs->data_layout. IIUC, the "layout" stored in var_layout_map represents the transposition from identity layout, while the "layout" in attr represents the layout in common sense, like NCHW. I suggest using the word "transposition" and "var_transposition_map" might be better.

Reply:
I think they are all layouts, just have different names for axis

Reply:
Then you should document that they have different names for axis, because it's really confusing and it's the biggest obstacle to me understanding the code.